contrib/dracut/90zfs/zfs-generator.sh.in

```diff
@@ -15,8 +15,10 @@ type getarg >/dev/null 2>&1 || . "$dracutlib"
 [ "${root##zfs:}" = "${root}" -a "${root##ZFS=}" = "${root}" -a "$rootfstype" != "zfs" ] && exit 0
 
 rootfstype=zfs
-if echo "${rootflags}" | grep -q zfsutil ; then
+if echo "${rootflags}" | grep -Eq '^zfsutil$|^zfsutil,|,zfsutil$|,zfsutil,' ; then
     true
+elif test -n "${rootflags}" ; then
+    rootflags="zfsutil,${rootflags}"
 else
     rootflags=zfsutil
 fi
```

Comment:
I've only tested on my Gentoo install (systemd 229 & dracut 044).  I think this is the first commit where systemd-fstab-generator started creating root mounts (https://github.com/systemd/systemd/commit/5e398e546ea65751e6a774daf828fe06f74434a2).  There and in the current code, it doesn't create a mount if there is no root option on the kernel command line.

Dracut may add it's own root option in the initrd, but I use the --no-hostonly-cmdline to strip that off.

That snippet goes just after the if statement that this comment thread belongs to in zfs-generator, just before the two string manipulation commands to remove zfs:/ZFS:.